sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
         return join;
     }
 
+
+    /**
+     * Build outputs of joins not in the outer level of the tree
```

Comment:
Maybe rephrase a bit. Something like "Extends the outputs of a querySpec to include symbols which are required by the next/upper join in the tree. These are queries that are not selected, but are for example used in a joinCondotion"